Neuro/Recording/LocalPlayer/LocalPlayerRecorder.cs

```diff
@@ -60,13 +61,28 @@ private void FixedUpdate()
 
             Frame.RaycastObstacleDistances[i] = raycastHit.distance;
         }
+
+        Frame.IsDead = PlayerControl.LocalPlayer.Data.IsDead;
+        RecordRole();
```

Comment:
Instead of calculating every frame, why not use the `OnGameStarted` to get the starting role, and `PlayerControl.Die` patch to record when they die and what role they become?